packages/marketo-form/partials/fields/checkbox-field/index.tsx

```diff
@@ -0,0 +1,61 @@
+import { useFormContext, useFormState } from 'react-hook-form'
+import CheckboxInput from '@hashicorp/react-checkbox-input'
+import Wrapper from '../../wrapper'
+import { formattedLabel } from '../../../utils'
+import type { MarketoFormCheckboxField } from '../../../types'
+
+const CheckboxField = ({ field }: { field: MarketoFormCheckboxField }) => {
+  const { register, watch } = useFormContext()
+  const { errors, touchedFields } = useFormState()
+  const value = watch(field.id, false)
+
+  if (field.id === 'Consent_Privacy_Policy__c') {
```

Comment:
💭 : This feels like a concern that shouldn't live in this component. I know you called it out in your earlier video so it's on your mind, but just noting that here. I don't think we need to address it immediately, but something to keep tabs on.

Reply:
Yeah, so there's three ways to add this functionality: in the base form field component as shown here, in the disambiguation component (as demonstrated [here](https://github.com/hashicorp/react-components/pull/630/files#diff-34895b2691e11077be55bd2655a9652901f961bcc6ce6c653a649b95c66605a1R34)), and finally as a default group, for example:

```ts
const defaultFieldGroupings = {
  name: {
    fields: ['FirstName', 'LastName'],
    component: NameField,
  },
  privacy: {
    fields: ['Consent_Privacy_Policy__c'],
    component: PrivacyCheckbox
  }
}
```

I'd prefer one of the first two over the third, since it mixes concepts of default field controls with default field groupings. However I agree with you that it doesn't feel quite right embedding that logic into the base form control component. The second approach (that the country combo box is using) feels slightly better, but it also still doesn't feel like the _right_ place.

Reply:
The disambiguation seems like a reasonable starting point to me! Agreed it doesn't feel fully _right_, but I think it's a reasonable starting point! 👍